beheer.md

```diff
@@ -104,13 +104,13 @@ docker service ps getstartedlab_web | Haal de gegevens op van alle tasks die de
 docker stack rm getstartedlab | Afsluiten van de applicatie.
 docker swarm leave --force | Afsluiten van de swarm.
 
-**Criteria**
-
-* API specificatie (OAS3) moet beschikbaar zijn en uit de imvertor genereerbaar zijn
-* Documentatie (zowel technisch als functioneel) moet beschikbaar zijn
-* De referentie-implementatie moet gereed en up-to date zijn met dan geldende OAS3
-* Het gebruikte Informatiemodel (minimaal UML)
-* Geautomatiseerde test en buildstraat (CI) moet beschikbaar voor de RI
-* De standaard moet ingebouwd zijn bij leveranciers (3 of meerdere leveranciers)
-* De standaard bevat een tabel waarin wordt aangegeven welke API versies met welke andere API versies compatible zijn.
+##Criteria##
+
+- API specificatie (OAS3) moet beschikbaar zijn en uit de imvertor genereerbaar zijn. Dit betekent dat zowel de Semantische Informatiemodellen (SIM), Uitwisselings Gegevensmodellen (UGM) als het Bericht Specificatiemodel (BSM) in UML beschikbaar moeten zijn.
```

Comment:
Van wie moet dit?
Voor wie moet dit?
Welk doel dient het?

Reply:
Op zich is het voor documentatie-doeleinden niet verkeerd om de objecten (zonder attributen?) en onderlinge relaties uit de API in een UML model te hebben. Dat zou als toevoeging aan de functionele documentatie kunnen dienen.

Even een beetje advocaat van de duivel spelend: als we vanuit beheer geen aanpassingen aan de API en referentiecomponenten gaan doorvoeren (maar dit laten doen), dan zou dit optioneel kunnen zijn. Voor die API standaarden die wel met UML gemodelleerd en gegenereerd zijn vanuit Imvertor moeten SIM, UGM en BSM wel beschikbaar zijn. Maar dat is omdat het dan eigenlijk de broncode (ruim zien ... :-) ) van de API is.  API standaarden die vanuit een referentie-implementatie gegenereerd worden (of is het eigenlijk een soort exporteren?) is dat dan helemaal niet nodig.

Maar, dit is technisch. Het beschikbaar hebben van de verschillende modellen kan wel meer inzicht geven in onderlinge relaties tussen resources en geeft mogelijk meer inzicht in de API. Zo gezien zou ik een globaal UML model van een API (resources + relaties, attributen haal je uit de OAS) wel een verplicht onderdeel van de documentatie vinden.

Andere insteek: zijn er leveranciers die vanuit UML code genereren? Dus niet vanuit de OAS maar vanuit een UML model? Dan heb je wel tools als rational rose en zo nodig denk ik maar het kan.

Als er vanuit de gebruikers (ontwikkelaars) behoefte aan deze modellen is moeten we het meenemen. Anders kan het optioneel blijven denk ik.

Reply:
Dan moet optioneel er bij.

Mij lijkt het geen voorwaarde om een OAS, documentatie en Docker in beheer te nemen. Imvertor is een manier om tot een OAS te komen maar geen verplichting.